app/src/main/java/ch/epfl/sdp/lobby/game/GameLobbyActivity.kt

```diff
@@ -0,0 +1,98 @@
+    }
+
+    private lateinit var mSwipeRefreshLayout : SwipeRefreshLayout
+    private lateinit var rv : RecyclerView
+    private var adminId = 0
+    private val repository = MockGameLobbyRepository
+    private lateinit var gameLobbyBinding: ActivityGameLobbyBinding
+    private var myFaction: PlayerParametersFragment.Faction = PlayerParametersFragment.Faction.PREDATOR
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        gameLobbyBinding = ActivityGameLobbyBinding.inflate(layoutInflater)
+        setContentView(gameLobbyBinding.root)
+
+        rv = findViewById(R.id.player_list)
+        rv.layoutManager = LinearLayoutManager(this)
+
+        //repository interactions
+        repository.setPlayerFaction(PLAYER_ID,PlayerParametersFragment.Faction.PREDATOR)
+        repository.getAdminId { id -> adminId = id }
+        repository.getPlayers { playerList -> rv.adapter = GameLobbyAdapter(playerList, PLAYER_ID,adminId) }
+
+        //set game info views
+        val gameInfo = findViewById<LinearLayout>(R.id.game_info)
+        repository.getGameName { name ->
+            val text = getText(R.string.game_name).toString() + " " + name
+            (gameInfo.getChildAt(0) as TextView).text = text}
+        repository.getGameDuration { game_duration ->
+            val text = getText(R.string.game_duration).toString() + " " + game_duration + " minutes"
+            (gameInfo.getChildAt(1) as TextView).text = text}
+
+        supportFragmentManager.beginTransaction().add(R.id.faction_selection,PlayerParametersFragment()).commit()
+
+        //SwipeRefreshLayout
+        mSwipeRefreshLayout = findViewById(R.id.swipe_container)
+        mSwipeRefreshLayout.setOnRefreshListener(this)
+
+        //add intents to the buttons
+        gameLobbyBinding.startButton.setOnClickListener {
+            val intent = if (myFaction == PlayerParametersFragment.Faction.PREDATOR) {
+                Intent(this, PredatorActivity::class.java)
+            } else {
+                Intent(this, PreyActivity::class.java)
+            }
+            startActivity(intent)
+        }
+        gameLobbyBinding.startButton.setBackgroundColor(Color.GREEN)
+
+        gameLobbyBinding.leaveButton.setOnClickListener {
+            startActivity(Intent(this, MainActivity::class.java))
+            startActivity(intent)
+        }
+        gameLobbyBinding.leaveButton.setBackgroundColor(Color.RED)
+
+    }
+
+    override fun onRefresh() {
+        refreshPlayerList()
+    }
+
+    override fun onFactionChange(newFaction: PlayerParametersFragment.Faction) {
+        //player id is hardcoded for now
+        repository.setPlayerFaction(PLAYER_ID,newFaction)
+        myFaction = newFaction
+    }
+
+    private fun refreshPlayerList() {
+        mSwipeRefreshLayout.isRefreshing = true
+        repository.getPlayers { playerList -> rv.adapter = GameLobbyAdapter(playerList, PLAYER_ID,adminId) }
```

Comment:
The end of refresh should be in the callback to account for real db's latency

```suggestion
        repository.getPlayers { playerList -> rv.adapter = GameLobbyAdapter(playerList, PLAYER_ID,adminId) 
        mSwipeRefreshLayout.isRefreshing = false
        }
```